doc/source/configuration/firewall.rst

```diff
@@ -0,0 +1,350 @@
+
+   # A firewalld zone to set as the default. Default is unset, in which case
+   # the default zone will not be changed.
+   # Predefined zones are listed here:
+   # https://firewalld.org/documentation/zone/predefined-zones.html
+   infra_vm_firewalld_default_zone: trusted
+
+   # A list of firewall rules to apply. Each item is a dict containing
+   # arguments to pass to the firewalld module. Arguments are omitted if not
+   # provided, with the following exceptions:
+   # - offline: true
+   # - permanent: true
+   # - state: enabled
+   infra_vm_firewalld_rules: "{{ stackhpc_firewalld_rules }}"
+
+Seed firewalld Configuration
+----------------------------
+
+.. code-block:: yaml
+   :caption: ``etc/kayobe/seed.yml``
+
+   ###############################################################################
+   # seed node firewalld configuration.
+
+   # Whether to install and enable firewalld.
+   seed_firewalld_enabled: true
+
+   # A list of zones to create. Each item is a dict containing a 'zone' item.
+   seed_firewalld_zones: "{{ stackhpc_firewalld_zones }}"
+
+   # A firewalld zone to set as the default. Default is unset, in which case
+   # the default zone will not be changed.
+   # Predefined zones are listed here:
+   # https://firewalld.org/documentation/zone/predefined-zones.html
+   seed_firewalld_default_zone: trusted
+
+   # A list of firewall rules to apply. Each item is a dict containing
+   # arguments to pass to the firewalld module. Arguments are omitted if not
+   # provided, with the following exceptions:
+   # - offline: true
+   # - permanent: true
+   # - state: enabled
+   seed_firewalld_rules: "{{ stackhpc_firewalld_rules }}"
+
+Seed Hypervisor firewalld Configuration
+---------------------------------------
+
+.. code-block:: yaml
+   :caption: ``etc/kayobe/seed_hypervisor.yml``
+
+   ###############################################################################
+   # seed_hypervisor node firewalld configuration.
+
+   # Whether to install and enable firewalld.
+   seed_hypervisor_firewalld_enabled: true
+
+   # A list of zones to create. Each item is a dict containing a 'zone' item.
+   seed_hypervisor_firewalld_zones: "{{ stackhpc_firewalld_zones }}"
+
+   # A firewalld zone to set as the default. Default is unset, in which case
+   # the default zone will not be changed.
+   # Predefined zones are listed here:
+   # https://firewalld.org/documentation/zone/predefined-zones.html
+   seed_hypervisor_firewalld_default_zone: trusted
+
+   # A list of firewall rules to apply. Each item is a dict containing
+   # arguments to pass to the firewalld module. Arguments are omitted if not
+   # provided, with the following exceptions:
+   # - offline: true
+   # - permanent: true
+   # - state: enabled
+   seed_hypervisor_firewalld_rules: "{{ stackhpc_firewalld_rules }}"
+
+Custom rules
+------------
+
+Custom firewalld rules can be added for any of the following groups using their
```

Comment:
```suggestion
Custom firewalld rules can be added for any of the following groups within an environment's ``group_vars`` inventory, ``etc/kayobe/environments/<environment>/inventory/group_vars/<group>/firewall`` using their
```

Reply:
They don't actually have to be in group vars. Host vars, group vars, and extra vars are all valid. I'll update the doc to make that clearer

Reply:
Done